common/src/contract.ts

```diff
@@ -163,14 +165,34 @@ export type Uniswap2 = {
   price: number
 }
 
+/**
+ * Implemented as a set of cpmm-1 binary contracts, one for each answer.
+ * The mechanism is stored among the contract's answers, which each
+ * reference this contract id.
+ */
+export type CPMMMulti = {
+  mechanism: 'cpmm-multi-1'
+  outcomeType: 'MULTIPLE_CHOICE'
+  shouldAnswersSumToOne: boolean
+  totalLiquidity: number // for historical reasons, this the total subsidy amount added in Ṁ
+  subsidyPool: number // current value of subsidy pool in Ṁ
+
+  // Answers chosen on resolution, with the weights of each answer.
+  // Weights sum to 1 if shouldAnswersSumToOne is true. Otherwise, range from 0 to 1 for each answerId.
+  resolutions?: { [answerId: string]: number }
```

Comment:
What about resolving N/A? If we want to handle conditional markets in the future, we'll need to be able to cancel some subset of the markets.